qiskit_experiments/calibration/fine_amplitude.py

```diff
@@ -0,0 +1,264 @@
+
+        This method can be overridden by subclasses e.g. to calibrate schedules on
+        transitions other than the 0 <-> 1 transition.
+        """
+        circuit = QuantumCircuit(1)
+
+        if self.experiment_options.add_sx:
+            circuit.sx(0)
+
+        if self.experiment_options.sx_schedule is not None:
+            sx_schedule = self.experiment_options.sx_schedule
+            circuit.add_calibration("sx", (self.physical_qubits[0],), sx_schedule, params=[])
+            circuit.barrier()
+
+        return circuit
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the fine amplitude calibration experiment.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            A list of circuits with a variable number of gates. Each gate has the same
+            pulse schedule.
+
+        Raises:
+            CalibrationError:
+                - If no schedule was provided.
+                - If the channel index does not correspond to the physical qubit index.
+                - If the schedule contains unassigned parameters.
+                - If the analysis options do not contain the angle_per_gate.
+        """
+
+        # Get the schedule and check assumptions.
+        schedule = self.experiment_options.get("schedule", None)
+
+        if schedule is None:
+            raise CalibrationError("No schedule set for fine amplitude calibration.")
+
+        if self.physical_qubits[0] not in set(ch.index for ch in schedule.channels):
+            raise CalibrationError(
+                f"User provided schedule {schedule.name} does not contain a channel "
+                "for the qubit on which to run the fine amplitude calibration."
+            )
+
+        if len(schedule.parameters) > 0:
+            raise CalibrationError(
+                "All parameters in a fine amplitude calibration schedule must be bound. "
+                f"Unbound parameters: {schedule.parameters}"
+            )
+
+        # Prepare the circuits.
+        gate = Gate(name=schedule.name, num_qubits=1, params=[])
+
+        repetitions = self.experiment_options.get("repetitions")
+
+        circuits = []
+
+        if self.experiment_options.add_xp_circuit:
+            angle_per_gate = self.analysis_options.get("angle_per_gate", None)
+            phase_offset = self.analysis_options.get("phase_offset")
+
+            if angle_per_gate is None:
+                raise CalibrationError(
+                    f"Unknown angle_per_gate for {self.__class__.__name__}. "
+                    "Please set it in the analysis options."
+                )
+
+            circuit = QuantumCircuit(1)
+            circuit.x(0)
+            circuit.measure_all()
+
+            circuit.metadata = {
+                "experiment_type": self._type,
+                "qubits": (self.physical_qubits[0],),
+                "xval": (np.pi - phase_offset) / angle_per_gate,

```

Comment:
Hmm, this isn't quite right, but I don't think the right way to fit this data point can be captured with the `CurveAnalysis` class. We want this point to fit to just `amp/2 + baseline`. With the way things are here, rotation error in this xval will be overweighted when calibrating xp because it will be treated as a half pulse instead of a full pulse. Perhaps we just overlook this since the qubit population is insensitive to rotation error to first order for an xp pulse, so this point won't contribute much to inferring the angle error any way? Perhaps a comment here about this is sufficient? And this feature won't be used outside of calibrating a pi pulse except...what happens if someone calibrates a pulse with `angle_per_gate==0`? :slightly_smiling_face:

Reply:
Good point. For now I think we can leave as is. When running this experiment on hardware I see that it nicely improves the pulse amplitude. It thus looks like this point is not contributing too the fit result. I left a comment here: d3fea7a